src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsics.Fixer.cs

```diff
@@ -0,0 +1,140 @@
+
+            if (model.GetOperation(node, context.CancellationToken) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            foreach (Diagnostic diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Properties.TryGetValue(nameof(RuleKind), out string? ruleKindName))
+                {
+                    continue;
+                }
+
+                if (!Enum.TryParse(ruleKindName, out RuleKind ruleKind))
+                {
+                    continue;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        UseCrossPlatformIntrinsicsTitle,
+                        c => ReplaceWithCrossPlatformIntrinsicAsync(context.Document, ruleKind, invocation, c),
+                        equivalenceKey: nameof(UseCrossPlatformIntrinsicsFixer)
+                    ),
+                    diagnostic
+                );
+            }
+        }
+
+        private async Task<Document> ReplaceWithCrossPlatformIntrinsicAsync(Document document, RuleKind ruleKind, IInvocationOperation invocation, CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+
+            SyntaxNode? replacementNode = ruleKind switch
+            {
+                RuleKind.opAddition => ReplaceBinaryOp(invocation, isCommutative: true, generator.AddExpression),
+                RuleKind.opBitwiseAnd => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseAndExpression),
+                RuleKind.opBitwiseOr => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseOrExpression),
+                RuleKind.opDivision => ReplaceBinaryOp(invocation, isCommutative: false, generator.DivideExpression),
+                RuleKind.opExclusiveOr => ReplaceBinaryOp(invocation, isCommutative: true, CreateExclusiveOrExpression),
+                RuleKind.opLeftShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateLeftShiftExpression),
+                RuleKind.opMultiply => ReplaceBinaryOp(invocation, isCommutative: true, generator.MultiplyExpression),
+                RuleKind.opOnesComplement => ReplaceUnaryOp(invocation, generator.BitwiseNotExpression),
+                RuleKind.opRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateRightShiftExpression),
+                RuleKind.opSubtraction => ReplaceBinaryOp(invocation, isCommutative: false, generator.SubtractExpression),
+                RuleKind.opUnaryNegation => ReplaceUnaryOp(invocation, generator.NegateExpression),
+                RuleKind.opUnsignedRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateUnsignedRightShiftExpression),
+                _ => null,
+            };
+
+            if (replacementNode is not null)
```

Comment:
This, for example, is problematic.  As it means the user gets offered to fix.  but then invoking the fix does nothing.

we consider this inappropriate behavior for an analyzer/fixer pair, and we want all the checking needed to ensure success to happen in the analyzer.

Reply:
Will adjust.